src/fu-util-common.c

```diff
@@ -27,6 +27,43 @@ fu_util_get_systemd_unit (void)
 	return SYSTEMD_FWUPD_UNIT;
 }
 
+gboolean
+fu_util_using_correct_daemon (GError **error)
+{
+	g_autoptr(GDBusConnection) connection = NULL;
+	g_autoptr(GVariant) val_path = NULL;
+	g_autoptr(GError) error_local = NULL;
+	const gchar *target = fu_util_get_systemd_unit ();
+
+	connection = g_bus_get_sync (G_BUS_TYPE_SYSTEM, NULL, error);
+	if (connection == NULL)
+		return FALSE;
+
+	val_path = g_dbus_connection_call_sync (connection,
+						SYSTEMD_SERVICE,
+						SYSTEMD_OBJECT_PATH,
+						SYSTEMD_MANAGER_INTERFACE,
+						"GetUnit",
+						g_variant_new ("(s)",
+								target),
+						NULL,
+						G_DBUS_CALL_FLAGS_NONE,
+						-1,
+						NULL,
+						&error_local);
+	if (val_path == NULL) {
+		g_debug ("%s", error_local->message);
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_INVALID_ARGS,
+			     /* TRANSLATORS: error message */
+			     _("wrong daemon is in use for this frontend, expected: %s"), target);
```

Comment:
Error messages are not normally translated -- although I guess this is a client tool helper and can be... but I really don't like the string itself. Maybe something like "Mismatched daemon and client versions, use /usr/bin/fwupdmgr instead" -- i.e. tell the user what they should be doing rather than a cryptic string they're going to google.

Reply:
I don't want to make inferences about the order of `PATH`, but I made some adjustments per your recommendations.